Investigate workflow job failing on main: typecheck / typecheck

🚨 **Failure Summary** 🚨:

- **📋 Job Name**: [typecheck / typecheck](https://github.com/Expensify/App/actions/runs/8145303098/job/22261238012)
- **🔧 Failure in Workflow**: Process new code merged to main
- **🔗 Triggered by PR**: [PR Link](https://github.com/Expensify/App/pull/37122)
- **👤 PR Author**: @shahinyan11
- **🤝 Merged by**: @francoisl
- **🐛 Error Message**: 
 warning: Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: nick-fields/retry@v2. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.
failure: Already included file name '/home/runner/work/App/App/src/types/utils/asMutable.ts' differs from file name '/home/runner/work/App/App/src/types/utils/AsMutable.ts' only in casing.
failure: Process completed with exit code 2.


⚠️ **Action Required** ⚠️:

🛠️ A recent merge appears to have caused a failure in the job named [typecheck / typecheck](https://github.com/Expensify/App/actions/runs/8145303098/job/22261238012).
This issue has been automatically created and labeled with `Workflow Failure` for investigation. 

👀 **Please look into the following**:
1. **Why the PR caused the job to fail?**
2. **Address any underlying issues.**

🐛 We appreciate your help in squashing this bug!

The action failed with:

```
Error: src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.tsx(24,23): error TS1261: Already included file name '/home/runner/work/App/App/src/types/utils/asMutable.ts' differs from file name '/home/runner/work/App/App/src/types/utils/AsMutable.ts' only in casing.
  The file is in the program because:
    Imported via '@src/types/utils/asMutable' from file '/home/runner/work/App/App/src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.tsx'
    Matched by include pattern 'src' in '/home/runner/work/App/App/tsconfig.json'
    Matched by include pattern 'src' in '/home/runner/work/App/App/tsconfig.json'
```

There's already a conversation about this in Slack - https://expensify.slack.com/archives/C01GTK53T8Q/p1709578543945489

--- a/src/types/utils/AsMutable.ts
+++ b/src/types/utils/AsMutable.ts
@@ -1,5 +1,0 @@
-import type {Writable} from 'type-fest';
-
-const asMutable = <T>(value: T): Writable<T> => value as Writable<T>;
-
-export default asMutable;

--- a/tests/utils/PusherHelper.ts
+++ b/tests/utils/PusherHelper.ts
@@ -3,7 +3,7 @@ import CONFIG from '@src/CONFIG';
 import CONST from '@src/CONST';
 import * as Pusher from '@src/libs/Pusher/pusher';
 import PusherConnectionManager from '@src/libs/PusherConnectionManager';
-import asMutable from '@src/types/utils/AsMutable';
+import asMutable from '@src/types/utils/asMutable';
 
 const CHANNEL_NAME = `${CONST.PUSHER.PRIVATE_USER_CHANNEL_PREFIX}1${CONFIG.PUSHER.SUFFIX}`;
 

